interfaces/builtin/bool_file.go

```diff
@@ -154,3 +154,12 @@ func (iface *BoolFileInterface) isGPIO(slot *interfaces.Slot) bool {
 func (iface *BoolFileInterface) AutoConnect() bool {
 	return false
 }
+
+// AutoConnectPair returns whether plug and slot should be implicitly
+// auto-connected assuming they will be an unambiguous connection
+// candidate and declaration-based checks allow.
+//
+// By default we allow what declarations allowed.
```

Comment:
I like these comments, but they are not included everywhere. It makes sense (to me anyway) to have the comment somewhere common rather than in the interfaces. Perhaps remove here and keep the one in core.go?